### ISSUE TITLE
Product Layered Navigation filters return duplicates

### ISSUE THREAD

**Issue by misha-kotov** [magento/graphql-ce#258]
The `Filters` object returned by the Products query should contain the available layered navigation filters array. If the Products query is filtered by category id, that is analogous to the shopper browsing to a category page, so returned GraphQL product filters in layered nav should correspond to the filters shoppers see on the category page. 

### Preconditions (*)

1. Magento 2.3
2. Sample data

### Steps to reproduce (*)

Issue GraphQL query
```
{
  products(filter: {category_id: {eq: "28"}}) {
    total_count
    filters {
      name
      filter_items_count
      filter_items{
        label
        value_string
        items_count
      }
    }
  }
}
```
### Expected result (*)

Returned layered navigation filters are the same as what is shown on the category page, **for both search engine implementations: MySQL and ElasticSearch**
![image](https://user-images.githubusercontent.com/18131925/48784771-f7272a00-eca8-11e8-8da7-004beaaeddd7.png)

### Actual result (*)

1. Layered navigation filters contain duplicates and do not correspond to what is shown on the category page. 
![image](https://user-images.githubusercontent.com/18131925/48784673-c1824100-eca8-11e8-8b28-d887a3c41463.png)

**Comment by heerdt** [comment]
Hello! 

Do you guys have any idea when this issue will be fixed?

I am facing the same problem described above, which the filters, returned from GraphQL, are not the same in the default Magento result page. 

### **Preconditions (*)**
Magento 2.3.2
Sample data

### **Steps to reproduce (*)**
Issue GraphQL query

```
{
  products(filter: {category_id: {eq: "12"}}) {
    total_count
    filters {
      name
      filter_items_count
      filter_items{
        label
        value_string
        items_count
      }
    }
  }
}
```

### **Expected result (*)**
Returned layered navigation filters are the same as what is shown on the category page, for both search engine implementations: MySQL and ElasticSearch

![image](https://user-images.githubusercontent.com/819013/63085329-2f604e00-bf45-11e9-92f8-5a901057d342.png)

### **Actual result (*)**
Layered navigation filters do not correspond to what is shown on the category page.

![image](https://user-images.githubusercontent.com/819013/63085408-60d91980-bf45-11e9-8981-e2c182131d51.png)

The returned facets are not using the query/filter applied to return the product list.

This is a blocker to build a Search result page or category page using GraphQL.

Thank you

**Comment by jayankaghosh** [comment]
Is there any update on this? This I feel is a very serious bug and renders the entire layered navigation useless.

**Comment by lenaorobei** [comment]
@TomashKhamlai please verify this on `2.3-develop` branch. It should be fixed by internal team for 2.3.4 release. Thanks.

**Comment by TomashKhamlai** [comment]
Still reproducible.
